### PR TITLE
[RS][DDCE-1213] Added status tag checks for section 8.

### PR DESCRIPTION
**Pull request by RafSobol**
# DDCE-(StoryNumberHere)

##Bug fix / New feature (delete as appropriate)

Please include a summary / description of the change and which issue it fixes.  Include any relevant user needs, context or links to other PRs related to this PR (eg. acceptance tests, environment config).

## PR Suggestions
- Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?
- Have you done a visual check of the changes?
- Is there any still commented or unused code? Lingering printlns?
- Have things been implemented in a complicated way?
- Are the test still over the coverage threshold?
- Does the compiler throw a lot of warnings? 
- Have methods and tests been named clearly?
- Were there any changes in the config? Do changes need to be made in app-config-???

## Checklist PR Raiser
 - [ ]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

## Checklist PR Reviewer
 - [x]  I've made every effort to commit high quality, clean code and I have executed relevant static analyses to be sure
 - [ ]  I've included appropriate tests with any code I've added (Unit, Integration, Acceptance etc.)
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [x]  I've squashed my commits - including the JIRA issue number in the commit message
 - [ ]  I've run a dependency check to ensure all dependencies are up to date

### CODE REVIEW

File: app/viewmodels/package.scala [hunk 1]
```diff
@@ -15,37 +15,50 @@
...
         updateList(
           isNino,
-          authorisedOfficial2common ++ Seq(AuthorisedOfficialsNinoPage(1)),
-          authorisedOfficial2common ++ Seq(AuthorisedOfficialsPassportPage(1))
+          commonPages ++ getPage(sectionPage, "nino", index),
+          commonPages ++ getPage(sectionPage, "passport", index)
         )
       }
 
+    def getOfficial(index: Int, isNino: Boolean, isPreviousAddress: Boolean, previousPages: Seq[QuestionPage[_]] = Seq.empty): Seq[QuestionPage[_]] = {
```
Comment (abhijeetardale) [inline app/viewmodels/package.scala:57]: getOfficialPages or getOfficialJourneyPages could be more readable

File: test/viewModels/otherOfficials/OtherOfficialStatusHelperSpec.scala [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+
+  def commonAllThree: UserAnswers = common(2, commonFirstTwo)
+
+  private val commonData = (index: Int, userAnswers: UserAnswers) => common(index, userAnswers)
+
+
+
+  "OtherOfficialsStatusHelper" must {
+
+    "when verifying section 7 answers" must {
```
Comment (abhijeetardale) [inline test/viewModels/otherOfficials/OtherOfficialStatusHelperSpec.scala:55]: please change the test for section 8

File: test/viewModels/otherOfficials/OtherOfficialStatusHelperSpec.scala [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+          _.set(IsOtherOfficialsPreviousAddressPage(0), false)).flatMap(
+          _.set(IsOtherOfficialNinoPage(1), true)).flatMap(
+          _.set(OtherOfficialsNinoPage(1), "AA123456A")).flatMap(
+          _.set(IsOtherOfficialsPreviousAddressPage(1), false)).flatMap(
+          _.set(AddAnotherOtherOfficialPage, true)).success.value) mustBe false
+      }
+
+      "return false with incorrect data for three other officials" in {
+        helper.checkComplete(commonAllThree.set(IsOtherOfficialNinoPage(0), false).flatMap(
+          _.set(OtherOfficialsPassportPage(0), Passport("GB12345", "GB", LocalDate.of(year, month, dayOfMonth)))).flatMap(
```
Comment (abhijeetardale) [inline test/viewModels/otherOfficials/OtherOfficialStatusHelperSpec.scala:345]: Here in this class lot of step are repeating for for creating user data. 
Could it be possible to define complete data set with 2 other officials and 3 other officials, then removing the data from as per the test requirement.

**Review by abhijeetardale — CHANGES_REQUESTED** [review]

File: test/viewModels/otherOfficials/OtherOfficialStatusHelperSpec.scala (hunk 3 above)
Comment (RafSobol) [reply test/viewModels/otherOfficials/OtherOfficialStatusHelperSpec.scala:345]: on it.

**Pull request merged**